invalidPropsals.ts

```diff
@@ -0,0 +1,145 @@
+                        }
+                    } else {
+                        console.log(proposal.id + " - is valid proposal")
+                    }
+                } catch(error) {
+                    console.log(error)
+                }
+            }));
+            await markProposalsInvalid(invalidProposalsInSpace);
+        }
+    } catch(error) {
+        console.log(error)
+    }
+}
+
+const verifyProposal = (proposal) => {
+    return new Promise(async (resolve, reject) => {
+        const space = spaces[proposal.space];
+        const result = {invalid: false, reason: ""}
+        if(!space) {
+            return reject(proposal.id + " - No Space found - " + proposal.space + " - Ignoring");
+        }
+        const members = space.members
+          ? space.members.map(address => address.toLowerCase())
+          : [];
+        const isMember = members.includes(proposal.address.toLowerCase());
+        
+        if (space.filters && space.filters.onlyMembers && !isMember) {
+            result.invalid = true;
+            result.reason = "not_by_member";
+            return resolve(result);
+        } else if (!isMember && space.filters && space.filters.minScore) {
+            try {
+                const scores = await snapshot.utils.getScores(
+                    proposal.space,
+                    space.strategies,
+                    space.network,
+                    snapshot.utils.getProvider(space.network),
+                    [proposal.address]
+                );
+                const totalScore: any = scores
+                    .map((score: any) =>
+                    Object.values(score).reduce((a, b: any) => a + b, 0)
+                    )
+                    .reduce((a, b: any) => a + b, 0);
+
+                if (totalScore < space.filters.minScore) {
+                    result.invalid = true;
+                    result.reason = "min_score";
+                    return resolve(result);
+                };
+            } catch (e) {
+                return reject(proposal.id + " - failed to check voting power - ignored");
+            }
+        }
+
+        return resolve(result);
+    });
+}
+
+const loadSpaces = async () => {
+    try {
+        const data = await fetch(spacesURL);
+        return data.json();
+    } catch(error) {
+        console.log(error);
+    }
+}
+
+const markProposalsInvalid = async (invalidProposalsInSpace) => {
+    try {
+        if(invalidProposalsInSpace.length > 0) {
+            const query = `UPDATE ${tableName} SET type = ? WHERE id IN (?)`;
+            await db.queryAsync(query, ['invalid-proposal', invalidProposalsInSpace ]);
+            console.log(invalidProposalsInSpace.length + ' proposals marked as invalid');
+        } else {
+            console.log("No proposals replaced in this space");
+        }
+    } catch(error) {
+        console.log(error);
+    }
+}
+
+const backup = async () => {

```

Comment:
We don't need this backup in the script. Technicaly the only thing that should change are message type from "proposal" to "invalid-proposal" that should be easy to revert if we see issue. The backup wont be usefull during the process of this script there will be new vote and proposals stored on the table so we wont be able to reuse the backup as it is.